aten/src/ATen/native/AveragePool3d.cpp

```diff
@@ -396,15 +398,15 @@ void avg_pool3d_backward_out_frame(
             }
 
             /* scatter gradients out to footprint: */
-            scalar_t val  = *op++;
+            const scalar_t val_scaled = *op++ / divide_factor;
 
             for (auto z = tstart; z < tend; z++)
             {
               for (auto y = hstart; y < hend; y++)
               {
                 for (auto x = wstart; x < wend; x++)
                 {
-                  *(ip + z * iheight * iwidth + y * iwidth + x) += val / divide_factor;
+                  *(ip + z * iheight * iwidth + y * iwidth + x) += val_scaled;
```

Comment:
This is a nice change, but have nothing to do with precision improvements, is it?